### PR TITLE
solve randomize_array and change run_test

### LINKED ISSUE #12: Challenge: Randomize Array
Solve brain_teasers/python/randomize_array/

Requirements:

write randomize array realization
add documentation
all tests are green

### PR DESCRIPTION
**Pull request by AnnaShliapkina**
Solve randomize_array, add README.md and add changes to run_tests to have possibility test only one package

- [x] PR's title includes the challenge's name and language
- [x] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [x] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [x] all conflicts are resolved (if any)
- [x] It is linked to an issue in the appropriate column of the project board
- [x] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [x] The markdown looks correct when you preview the file
- [ ] All links and images work
- [x] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [x] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [x] There is a module header
- [x] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [x] Behavior description
- [ ] Parameter description
- [x] Return value description
- [x] Include any assumptions
- [x] Include 3 or more (passing!) doctests
- [x] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [ ] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [ ] There are type annotations
- [ ] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Each unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
- [x] All tests pass
- [ ] There are tests for defensive assertions
- [x] There are tests for boundary cases
- [ ] Test suite includes black-box and glass-box tests

### CODE REVIEW

**Review by MajorPrestige — APPROVED** [review]
Brilliant!

**Review by DmytroLyt — APPROVED** [review]
Great work! 
Some small marks:
- The file's name is not matches the function name
- The test class without a docstring

File: src/brain_teasers/python/randomize_array/tests/test_randomize_array.py
Comment (MurtazaHakimi) [inline src/brain_teasers/python/randomize_array/tests/test_randomize_array.py:1]: Hi Anna, 
as I see you did so much correction. that means how in detail you have been worked.
I appreciate your work.

File: src/brain_teasers/python/randomize_array/tests/test_randomize_array.py [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
-from src.randomize_array import randomize_array
+import os
+import sys 
+
+file_dir = os.path.dirname(os.path.dirname(__file__))
+sys.path.append(file_dir + "/src")
+
+from  randomize_array import improved_fisher_yates_shuffle
 
+class TestShuffleFunction(unittest.TestCase):
```
Comment (MurtazaHakimi) [inline src/brain_teasers/python/randomize_array/tests/test_randomize_array.py:12]: this class is empty?

**Review by MurtazaHakimi — COMMENTED** [review]
Hi Anna, 
I have reviewed your work.

File: src/brain_teasers/python/randomize_array/tests/test_randomize_array.py (hunk 1 above)
Comment (AnnaShliapkina) [reply src/brain_teasers/python/randomize_array/tests/test_randomize_array.py:12]: No, there are a lot of functions but here you can see only changes in code, not hole file. But if push on the link above you can see hole file
